examples/medplum-task-demo/src/components/actions/AddNote.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button, Modal } from '@mantine/core';
+import { notifications } from '@mantine/notifications';
+import {
+  createReference,
+  getQuestionnaireAnswers,
+  MedplumClient,
+  normalizeErrorString,
+  PatchOperation,
+} from '@medplum/core';
+import { Annotation, Questionnaire, QuestionnaireResponse, Task } from '@medplum/fhirtypes';
+import { QuestionnaireForm, useMedplum, useMedplumProfile } from '@medplum/react';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useState } from 'react';
+
+interface AddCommentProps {
+  task: Task;
+  onChange: (updatedTask: Task) => void;
+}
+
+export function AddNote(props: AddCommentProps): JSX.Element {
+  const medplum = useMedplum();
+  const author = useMedplumProfile();
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
```

Comment:
Consider Mantine `useDisclosure` for these, might be a little cleaner: https://mantine.dev/hooks/use-disclosure/